server/finder/http/server.go

```diff
@@ -49,34 +49,22 @@ func New(listen string, indexer indexer.Interface, registry *registry.Registry,
 	h := newHandler(indexer, registry)
 
 	// Client routes
-	cidR := mux.NewRouter().StrictSlash(true)
-	cidR.HandleFunc("/cid/{cid}", h.findCid).Methods(http.MethodGet, http.MethodOptions)
-	corCidR := handlers.CORS(handlers.AllowedOrigins([]string{"*"}))(cidR)
-
-	mhR := mux.NewRouter().StrictSlash(true)
-	mhR.HandleFunc("/multihash/{multihash}", h.find).Methods(http.MethodGet, http.MethodOptions)
-	mhR.HandleFunc("/multihash", h.findBatch).Methods(http.MethodPost, http.MethodOptions)
-	corMhR := handlers.CORS(handlers.AllowedOrigins([]string{"*"}))(mhR)
-
 	r := mux.NewRouter().StrictSlash(true)
-	r.Use(mux.CORSMethodMiddleware(r))
-	r.Use(jsonContentTypeAndTerminateOnOptions)
-	r.PathPrefix("/cid").Handler(corCidR)
-	r.PathPrefix("/multihash").Handler(corMhR)
-
-	r.HandleFunc("/health", h.health).Methods(http.MethodGet, http.MethodOptions)
 	r.Handle("/", http.FileServer(http.FS(webUI)))
-
-	r.HandleFunc("/providers", h.listProviders).Methods(http.MethodGet, http.MethodOptions)
-	r.HandleFunc("/providers/{providerid}", h.getProvider).Methods(http.MethodGet, http.MethodOptions)
-
-	r.HandleFunc("/stats", h.getStats).Methods(http.MethodGet, http.MethodOptions)
+	r.HandleFunc("/cid/{cid}", h.findCid).Methods(http.MethodGet)
+	r.HandleFunc("/multihash/{multihash}", h.find).Methods(http.MethodGet)
+	r.HandleFunc("/multihash", h.findBatch).Methods(http.MethodPost)
+	r.HandleFunc("/health", h.health).Methods(http.MethodGet)
+	r.HandleFunc("/providers", h.listProviders).Methods(http.MethodGet)
+	r.HandleFunc("/providers/{providerid}", h.getProvider).Methods(http.MethodGet)
+	r.HandleFunc("/stats", h.getStats).Methods(http.MethodGet)
 
 	reframeHandler := reframe.NewReframeHTTPHandler(indexer, registry)
 	r.HandleFunc("/reframe", reframeHandler)
 
+	cors := handlers.CORS(handlers.AllowedOrigins([]string{"*"}))
 	server := &http.Server{
-		Handler:      r,
+		Handler:      cors(r),
```

Comment:
@gammazero I'd love your eagle eye on the changes here. I think the changes I have made simplifies the way mux router is set up while keeping the intended functionality the same. Many thanks

Reply:
I do not know why we did not do this in the first place.  I am guessing we thought we did not want to have CORS for all endpoints.  Anyway, this is much cleaner and looks correct.